NachoSpigot-Server/src/main/java/dev/cobblesword/nachospigot/hitdetection/LagCompensator.java

```diff
@@ -37,12 +37,13 @@ public Location getHistoryLocation(int rewindMillisecs, Player player) {
         long currentTime = System.currentTimeMillis();
 
         int rewindTime = rewindMillisecs + pingOffset;
+        int timesSize = times.size();
```

Comment:
One small thing, you never use `timesSize`, you always use `timesSize - 1`. I'd recommend changing this line to `int timesSize = times.size() - 1` and remove the calculations in the for loop.

Reply:
Not sure how I missed that but I just fixed it. Thanks for letting me know about it